test/functional/wallet_avoidreuse.py

```diff
@@ -240,5 +242,39 @@ def test_fund_send_fund_send(self):
         assert_approx(self.nodes[1].getbalance(), 1, 0.001)
         assert_approx(self.nodes[1].getbalance(avoid_reuse=False), 11, 0.001)
 
+    def test_many_utxos_one_dest(self):
+        '''
+        Test the case where [1] generates a new address A, then
+        [0] sends 1 BTC to A 30 times.
+        [1] spends 11 BTC from A
+        20 of the outputs in A should be used, leaving 10 intact.
+        Those 10 should all be marked used.
+        A single output of 9 btc should be the change output back to oneself,
+        and should not be marked used.
+        '''
+        self.log.info("Test many utxos one destination")
+
+        fundaddr = self.nodes[1].getnewaddress()
+        retaddr = self.nodes[0].getnewaddress()
+
+        [self.nodes[0].sendtoaddress(fundaddr, 1) for _ in range(30)]
```

Comment:
Generally, this is not a pattern seen in the codebase. When list comprehensions are done, they're done to assign to a variable. It's also a little harder to read. A two-liner like this is easier.:

    for _ in range(30):
        self.nodes[0].sendtoaddress(fundaddr, 1)

Reply:
Also, a comment above would be useful ( `# send one bitcoin to the funding address 30 times`)

Reply:
No, there are cases where this is done

Reply:
OK, I missed those. I still don't think it's that clear, but if it's there already, I'm happy to have it be a pattern.

Reply:
Yeah, you're probably right. I'll change it to be more clear and add comment. Thanks for the feedback.